src/hooks/useIsWindowHeightReducedByKeyboard/index.ts

```diff
@@ -0,0 +1,31 @@
+import {useCallback, useEffect, useState} from 'react';
+import usePrevious from '@hooks/usePrevious';
+import useResponsiveLayout from '@hooks/useResponsiveLayout';
+import useWindowDimensions from '@hooks/useWindowDimensions';
+
+const useIsWindowHeightReducedByKeyboard = () => {
+    const [isWindowHeightReducedByKeyboard, setIsWindowHeightReducedByKeyboard] = useState(false);
+    const {windowHeight} = useWindowDimensions();
+    const prevWindowHeight = usePrevious(windowHeight);
+    const {shouldUseNarrowLayout} = useResponsiveLayout();
+    const toggleKeyboardOnSmallScreens = useCallback(
+        (isKBOpen: boolean) => {
+            if (!shouldUseNarrowLayout) {
+                return;
+            }
+            setIsWindowHeightReducedByKeyboard(isKBOpen);
+        },
+        [shouldUseNarrowLayout],
+    );
+    useEffect(() => {
+        if (!isWindowHeightReducedByKeyboard && windowHeight < prevWindowHeight - 100) {
```

Comment:
Can you add comments for this effect? Copying from PDFView/index.tsx should be fine.

Reply:
I just added comment